### DISCUSSION TITLE
Trouble power cycling usb microphone

### DISCUSSION THREAD

**Discussion by ddesimio** [mvp/uhubctl#398] [Q&A] [answered]
I have two hubs and want to power cycle one device on one of them because if I leave it plugged in on boot/reboot, it does not work.

$ sudo uhubctl
Current status for hub 6-4 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 1: 02a0 power 5gbps Rx.Detect
  Port 2: 02a0 power 5gbps Rx.Detect
  Port 3: 02a0 power 5gbps Rx.Detect
  Port 4: 02a0 power 5gbps Rx.Detect
Current status for hub 5-3 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 1: 0100 power
  Port 2: 0100 power
  Port 3: 0100 power
  Port 4: 0503 power highspeed enable connect [046d:0ab7 Generic Blue Microphones LT_2105111044219D030400_111000] <- THIS ONE
Current status for hub 2-2.1 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 1: 02a0 power 5gbps Rx.Detect
  Port 2: 02a0 power 5gbps Rx.Detect
  Port 3: 02a0 power 5gbps Rx.Detect
  Port 4: 02a0 power 5gbps Rx.Detect
Current status for hub 2-2 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 1: 0263 power 5gbps U3 enable connect [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 2: 02a0 power 5gbps Rx.Detect
  Port 3: 02a0 power 5gbps Rx.Detect
  Port 4: 02a0 power 5gbps Rx.Detect
Current status for hub 1-2.1 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 1: 0507 power highspeed suspend enable connect [046d:0894 1080P Pro Stream AEC19730]
  Port 2: 0100 power
  Port 3: 0103 power enable connect [088e:5036 iLok iLok 3FA0812A_640B3264]
  Port 4: 0503 power highspeed enable connect [1235:800a Focusrite Scarlett 2i4 USB]
Current status for hub 1-2 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 1: 0503 power highspeed enable connect [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 2: 0103 power enable connect [0819:0101 Syncrosoft eLicenser]
  Port 3: 0103 power enable connect [1b1c:1b5a Corsair CORSAIR M65 RGB ELITE Gaming Mouse 0F02D023AF5810A65D466A45F5001BC4]
  Port 4: 0103 power enable connect [1b1c:1b6b Corsair CORSAIR K70 RGB MK.2 SE Mechanical Gaming Keyboard 0F02A027AF6895215E231A37F5001BC6]
  
  I think the two hubs are [1XX, 2XX], and [6-4, 5-3] so to turn the Blue Microphone off then on I used:

  $ sudo uhubctl -a off -p 4 -l 5-3

and got:

Current status for hub 6-4 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 4: 02a0 power 5gbps Rx.Detect
Sent power off request
New status for hub 6-4 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 4: 0080 off
Current status for hub 5-3 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 4: 0503 power highspeed enable connect [046d:0ab7 Generic Blue Microphones LT_2105111044219D030400_111000]
Sent power off request
New status for hub 5-3 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 4: 0000 off

then:

$ sudo uhubctl -a on -p 4 -l 5-3
Current status for hub 6-4 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 4: 0080 off
Sent power on request
New status for hub 6-4 [2109:0813 VIA Labs, Inc. USB3.0 Hub, USB 3.00, 4 ports, ppps]
  Port 4: 02a0 power 5gbps Rx.Detect
Current status for hub 5-3 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 4: 0503 power highspeed enable connect [046d:0ab7 Generic Blue Microphones LT_2105111044219D030400_111000]
Sent power on request
New status for hub 5-3 [2109:2813 VIA Labs, Inc. USB2.0 Hub, USB 2.10, 4 ports, ppps]
  Port 4: 0503 power highspeed enable connect [046d:0ab7 Generic Blue Microphones LT_2105111044219D030400_111000]

My problem is that the Blue Microphone state never seems to change (unresponsive volume meter in pavucontrol) so I am wondering what I am doing wrong or maybe if I have an incompatible hub. 

will try again when this arraives: https://www.amazon.com/Plugable-USB-Charging-Laptops-USB-C/dp/B096YKDYX5/ref=sr_1_3?crid=14E2RM88FQW4B&keywords=plugable+USBC-HUB7BC&qid=1644945231&sprefix=plugable+usbc-hub7bc%2Caps%2C93&sr=8-3#customerReviews

**Comment by mvp** [comment, marked as answer]
You can check if your hub is working properly by attaching phone or USB light to hub port and turn it off - if phone stops charging or light stops shining, then it works properly. Many USB hubs will not actually cut power off (but most will cut off data lines).
All hubs in supported list should be turning off power (VBUS) for real.

**Reply by ddesimio** [reply]
works perfectly with the new hub: https://www.amazon.com/Plugable-USB-Charging-Laptops-USB-C/dp/B096YKDYX5/ref=sr_1_3?crid=14E2RM88FQW4B&keywords=plugable+USBC-HUB7BC&qid=1644945231&sprefix=plugable+usbc-hub7bc%2Caps%2C93&sr=8-3#customerReviews
